### PR TITLE
Remove workaround from `ArtifactRequestHandler`

### PR DESCRIPTION
**Pull request by DominikPinsel**
Signed-off-by: Dominik Pinsel <dominik.pinsel@daimler.com>

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

### CODE REVIEW

**Comment by DominikPinsel** [comment]
@jimmarino This pull request removes the workaround from the IDS `ArtifactRequestHandler`, so that the data is transferred using the real contract. We probably need this for the presentation this afternoon.

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java [hunk 1]
```diff
@@ -117,12 +118,11 @@ public NegotiationResponse initiate(ContractOfferRequest contractOffer) {
      * ContractNegotiation and transitions the ContractNegotiation to CONSUMER_APPROVING,
      * CONSUMER_OFFERING or DECLINING.
      *
-     * @param token Claim token of the consumer that send the contract request.
+     * @param token         Claim token of the consumer that send the contract request.
```
Comment (jimmarino) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java:121]: Can you just clean up the spacing change?

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java [hunk 2]
```diff
@@ -117,12 +118,11 @@ public NegotiationResponse initiate(ContractOfferRequest contractOffer) {
...
      * @param negotiationId Id of the ContractNegotiation.
      * @param contractOffer The contract offer.
-     * @param hash A hash of all previous contract offers.
-     * @return a {@link NegotiationResponse}: FATAL_ERROR, if no match found for Id or no last
-     *         offer found for negotiation; OK otherwise
+     * @param hash          A hash of all previous contract offers.
```
Comment (jimmarino) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java:124]: Can you just clean up the spacing change?

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java [hunk 3]
```diff
@@ -170,12 +170,11 @@ public NegotiationResponse offerReceived(ClaimToken token, String negotiationId,
      * Validates the contract agreement sent by the provider against the last contract offer and
      * transitions the corresponding {@link ContractNegotiation} to state CONFIRMED or DECLINING.
      *
-     * @param token Claim token of the consumer that send the contract request.
+     * @param token         Claim token of the consumer that send the contract request.
```
Comment (jimmarino) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java:173]: Can you just clean up the spacing change?

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java [hunk 4]
```diff
@@ -219,10 +218,9 @@ public NegotiationResponse confirmed(ClaimToken token, String negotiationId, Con
      * Tells this manager that a {@link ContractNegotiation} has been declined by the counter-party.
      * Transitions the corresponding ContractNegotiation to state DECLINED.
      *
-     * @param token Claim token of the consumer that sent the rejection.
+     * @param token         Claim token of the consumer that sent the rejection.
```
Comment (jimmarino) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java:221]: Can you just clean up the spacing change?

File: core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java [hunk 5]
```diff
@@ -243,7 +241,7 @@ public NegotiationResponse declined(ClaimToken token, String negotiationId) {
      * Builds and sends a {@link ContractOfferRequest} for a given {@link ContractNegotiation} and
      * {@link ContractOffer}.
      *
-     * @param offer The contract offer.
+     * @param offer   The contract offer.
```
Comment (jimmarino) [inline core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java:244]: Can you just clean up the spacing change?

**Review by jimmarino — APPROVED** [review]
Please just fix the small formatting errors

**Pull request merged**